### DISCUSSION TITLE
LibCore 2024.2.29

### DISCUSSION THREAD

**Discussion by Fasjeit** [CryptoPro/libcore#50] [Announcements]
Первый **бета** релиз, с возможностью сборки под 6.0 и 8.0 без исправлений рантайма.

[**Не является совместимым с прошлыми версиями!**](https://dss.cryptopro.ru/libcore/docs/03-%D0%BE%D1%81%D0%BE%D0%B1%D0%B5%D0%BD%D0%BD%D0%BE%D1%81%D1%82%D0%B8-%D1%80%D0%B0%D0%B1%D0%BE%D1%82%D1%8B/03-01-%D0%B8%D0%B7%D0%BC%D0%B5%D0%BD%D0%B5%D0%BD%D0%B8%D1%8F-api.html)

Подробности в документации:
https://dss.cryptopro.ru/libcore/

[В работе / не реализовано:](https://dss.cryptopro.ru/libcore/docs/03-%D0%BE%D1%81%D0%BE%D0%B1%D0%B5%D0%BD%D0%BD%D0%BE%D1%81%D1%82%D0%B8-%D1%80%D0%B0%D0%B1%D0%BE%D1%82%D1%8B/03-02-%D0%BD%D0%B5-%D1%80%D0%B5%D0%B0%D0%BB%D0%B8%D0%B7%D0%BE%D0%B2%D0%B0%D0%BD%D0%BE.html)
* SignedCms
* EnvelopedCms
* Установка исправлений на классы рантайма

<hr /><em>This discussion was created from the release <a href='https://github.com/CryptoPro/libcore/releases/tag/v2024.2.29'>LibCore 2024.2.29</a>.</em>

**Comment by Bykiev** [comment]
Спасибо за бета-версию, а что значит пункт "Установка исправлений на классы рантайма", если сама библиотека работает без исправления рантайма?

**Reply by Fasjeit** [reply]
Установка исправлений, если будет реализована, будет опциональной. 

Т.е. можно будет как пользоваться честной библиотечной частью, так и при желании исправленными родными классами.

**Reply by Bykiev** [reply]
Я ведь правильно понимаю, что без таких исправлений сторонние библиотеки не будут работать (тот же MimeKit)? В этом случае чем новая реализация будет отличаться от текущей (исправления через Harmony)?

**Reply by Fasjeit** [reply]
Да, сторонние либы в текущем релизе не заведутся с гостами.

В случае исправлений - ничем, будут точно такие же исправления, как в старой версии (внутри только устроенные иначе), но с поддержкой 8.0 и возможность отказаться от них, если они не нужны.

**Reply by Bykiev** [reply]
А не рассматривали возможность внести изменения непосредственно в .NET, тогда патчить классы самостоятельно не придется?

**Reply by Fasjeit** [reply]
Если речь про форк - то уже [пробовали](https://github.com/CryptoPro/corefx).

**Reply by Bykiev** [reply]
Не форкать, а внести изменения в репозиторий Microsoft

**Reply by Fasjeit** [reply]
Изначально такой план и был, но ms не заинтересованы в расширении криптографии рантайма (даже на уровне интерфейсов) для интеграции дополнительных алгоритмов.
